### PR TITLE
Base repository conductors for core project setup

### PR DESCRIPTION
**Pull request by dylanjustice**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  - DOTNET_CLI_TELEMETRY_OPTOUT: 1
+language: csharp
+mono: none
+notifications:
+  slack:
+    on_failure: always
+    on_pull_requests: true
+    on_success: always
+    rooms:
+      secure: Lb4zNaOgM7hF705GykSv8ibU2paT9O8RxJQUkek+U1uIenxL8h0uF3gIn+/dmWkxtsPsbCt1it5NzYbLRcH/waQg6/xJIMEtVve9S1EnCaP8gZid8xDccPTFNlJk5v9bfWSXmvf4k7ufY2thit0COcm9DT4WoW4HUlSA487r40zmyhzHaD2h0ojF9le7yIxiXEG/Ojgi7TcAW+WLZeF8Y9o3sR4vqLTHxRaZ/YhyrmwJ6CjYMLQWFSIuuSdYzyUYyL2sHBpY6xqyCC5KcYqHLDwSG2vNJ0WTaJ0W4tiKGHdglPvCW2MCZdF1ROfJbv4dk2xUHJkB8Y3AMfaI5EWJyf3WVwGjioOsImXKP7Ga5DT6YnKf3iWi3eILCry2527UacaiDIlMvLWyqvHb++Zp+TnFFp9xYrBG7aCumbE/+ASBMxIEcyi1BbqJKAtYLN5XWLpyZ2VdIpaadab8LBehz/hBsLXnT7swlk7ANWGVXBe7GuKnj+z8hg/n74NqwbIqhBYymCPHFlMr4+uIkPaXXuBGyCgDWUDqMupe0ff/8wXYcFmZSmgMvat2eQ96fb8wcKCLOoeN0Dcq59nc++z741g8Tb3PHnmb9D7lw9IIOUyrpP14q15wug4PFVtDgq6n7uACgzTS/BC8RAeGcYeJoFzfYT8Wt1LTuE0tx896XGs=
```
Comment (wintondeshong) [inline .travis.yml]: Just double checking that this was tested out and verified by using the travis-ci cli

File: package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git@github.com/AndcultureCode/AndcultureCode.CSharp.Business.Conductors.git"
+  },
+  "author": "andculture",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/AndcultureCode/AndcultureCode.CSharp.Core/issues"
```
Comment (wintondeshong) [inline package.json]: Minor, but update path to match 'Conductors' repo path `https://github.com/AndcultureCode/AndcultureCode.CSharp.Conductors/issues`

File: package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "dependencies": {
+    "and-cli": "0.0.5"
+  },
+  "devDependencies": {},
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git@github.com/AndcultureCode/AndcultureCode.CSharp.Business.Conductors.git"
```
Comment (wintondeshong) [inline package.json]: `Business` is not a part of the repo name

File: src/AndcultureCode.CSharp.Business.Conductors/RepositoryConductor.cs [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+    where T : Entity
+    {
+        #region Properties
+
+        public int? CommandTimeout
+        {
+            get => _readConductor.CommandTimeout;
+            set
+            {
+                _createConductor.CommandTimeout = value;
```
Comment (wintondeshong) [inline src/AndcultureCode.CSharp.Business.Conductors/RepositoryConductor.cs]: Minor, alignment

File: src/AndcultureCode.CSharp.Business.Conductors/RepositoryConductor.cs [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+                _deleteConductor.CommandTimeout = value;
+                _readConductor.CommandTimeout = value;
+                _updateConductor.CommandTimeout = value;
+            }
+        }
+
+        protected readonly IRepositoryCreateConductor<T> _createConductor;
```
Comment (wintondeshong) [inline src/AndcultureCode.CSharp.Business.Conductors/RepositoryConductor.cs]: Minor, alignment

File: src/AndcultureCode.CSharp.Business.Conductors/RepositoryConductor.cs [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+        #region Update
+
+        public virtual IResult<bool> BulkUpdate(IEnumerable<T> items, long? updatedBy = default(long?)) => _updateConductor.BulkUpdate(items, updatedBy);
+        public virtual IResult<bool> Update(T item, long? updatedBy = default(long?)) => _updateConductor.Update(item, updatedBy);
+        public virtual IResult<bool> Update(IEnumerable<T> items, long? updatedBy = default(long?)) => _updateConductor.Update(items, updatedBy);
+
+        #endregion Update
+
+
+        #region Delete
```
Comment (wintondeshong) [inline src/AndcultureCode.CSharp.Business.Conductors/RepositoryConductor.cs]: Recommend alphabetizing regions

File: test/AndcultureCode.CSharp.Business.Conductors.Tests/AndcultureCode.CSharp.Business.Conductors.Tests.csproj [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  <ItemGroup>
+    <PackageReference Include="AndcultureCode.CSharp.Core" Version="0.1.1" />
+    <PackageReference Include="AndcultureCode.CSharp.Testing" Version="0.0.2" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
+    <PackageReference Include="xunit" Version="2.4.0" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\..\src\AndcultureCode.CSharp.Business.Conductors\AndcultureCode.CSharp.Business.Conductors.csproj" />
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Business.Conductors.Tests/AndcultureCode.CSharp.Business.Conductors.Tests.csproj]: Update to match that `Business` isn't a part of the package name

File: test/AndcultureCode.CSharp.Business.Conductors.Tests/TestEntity.cs [hunk 8]
```diff
@@ -0,0 +1,10 @@
+using System;
+using AndcultureCode.CSharp.Core.Models.Entities;
+
+namespace AndcultureCode.CSharp.Business.Conductors.Tests
+{
+    public class TestEntity : Entity
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Business.Conductors.Tests/TestEntity.cs]: Is this a test stub? If so, recommend creating a `stubs` folder

File: test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs [hunk 9]
```diff
@@ -0,0 +1,493 @@
...
+
+        [Fact]
+        public void BulkCreateOrUpdate_When_BulkCreate_Has_Errors_Then_Result_Has_Errors_For_BulkCreate()
+        {
+            // Arrange
+            var mockUpdateConductor = new Mock<IRepositoryUpdateConductor<Entity>>();
+            var mockCreateConductor = new Mock<IRepositoryCreateConductor<Entity>>();
+
+            var entities = new List<Entity>();
+            var entity   = new TestEntity() { Id = 0, Name = "hello-world" };
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs]: Minor, `()` is not needed

File: test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs [hunk 10]
```diff
@@ -0,0 +1,493 @@
...
+                updateConductor: mockUpdateConductor
+            );
+
+            // Act
+            var bulkCreateOrUpdateResponse = sut.BulkCreateOrUpdate(entities.AsEnumerable(), 1);
+
+            // Assert
+            bulkCreateOrUpdateResponse.HasErrors.ShouldBeTrue();
+            bulkCreateOrUpdateResponse.Errors.FirstOrDefault().Key.ShouldBe(BASIC_ERROR_KEY);
+        }
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs]: Uber minor, newline pad between functions

File: test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs [hunk 11]
```diff
@@ -0,0 +1,493 @@
...
+                            Message = BASIC_ERROR_MESSAGE}
+                    },
+                ResultObject = null
+            });
+
+            var sut = SetupSut(
+                createConductor: mockCreateConductor,
+                updateConductor: mockUpdateConductor
+            );
+            // Act
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs]: Uber minor, newline pad above `// Act`

File: test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs [hunk 12]
```diff
@@ -0,0 +1,493 @@
...
+            var bulkCreateOrUpdateResponse = sut.BulkCreateOrUpdate(entities, 1);
+
+            // Assert
+            bulkCreateOrUpdateResponse.HasErrors.ShouldBeTrue();
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs]: Consider using the `IResultMatcherExtensions` `ShouldHaveErrorsFor<T>` matcher extension.

File: test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs [hunk 13]
```diff
@@ -0,0 +1,493 @@
...
+
+            var sut = SetupSut(
+                createConductor: mockCreateConductor,
+                updateConductor: mockUpdateConductor
+            );
+            // Act
+            var bulkCreateOrUpdateResponse = sut.BulkCreateOrUpdate(entities, 1);
+
+            // Assert
+            bulkCreateOrUpdateResponse.HasErrors.ShouldBeFalse();
```
Comment (wintondeshong) [inline test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs]: Recommend using the `IResultMatcherExtensions` `ShouldNotHaveErrors<T>` matcher

Comment (dylanjustice) [reply test/AndcultureCode.CSharp.Business.Conductors.Tests/RepositoryConductorTest.cs]: You're absolutely right. Completely thought this wasn't implemented yet.

**Comment by brandongregoryscott** [comment]
@all-contributors add @dylanjustice for code, test

**Comment by brandongregoryscott** [comment]
@all-contributors add @wintondeshong for review

**Pull request merged**